### PR TITLE
Fix for subcommands with @Command UDA without a name

### LINKED ISSUE #111: Not providing a program name requires to pass an empty string
> If program name is not provided, then `Runtime.args[0]` (a.k.a. `argv[0]` from `main` function) is used.

Here’s how we “not provide” a program name:

```d
import argparse;

@(Command("").Description("A program's description."))
struct Args {
    bool banana;
}

mixin CLI!Args.main!((in _) { });
```

`Command()` should work as well, shouldn’t it? But it doesn’t; it throws an `ArrayIndexError`.

### PR DESCRIPTION
**Pull request by andrey-zherikov**
Closes #111

### CODE REVIEW

File: source/argparse/internal/commandinfo.d [hunk 1]
```diff
@@ -58,10 +60,27 @@ package(argparse) CommandInfo getCommandInfo(COMMAND)(const Config config, strin
...
+    {
+        CommandInfo info = udas[0];
+
+        if(name.length > 0 && info.names.length == 0)
+            info.names = [name];
+
+        info = finalize(config, info);
+    }
     else
         CommandInfo info = finalize(config, CommandInfo([name]));
```
Comment (SirNickolas) [inline source/argparse/internal/commandinfo.d:72]: If `udas.length > 0 && name.length == 0 && info.names.length == 0`, `info.names` will be `[]`.
If `udas.length == 0 && name.length == 0 && info.names.length == 0`, `info.names` will be `[""]`.

I’m suspicous about such inconsistency. We should either drop `name.length > 0`, add similar logic to the no-UDA branch, or move this logic somewhere else.

Comment (andrey-zherikov) [reply source/argparse/internal/commandinfo.d]: `name` is set for subcommands only there. I did some refactoring and split `getCommandInfo` into `getSubCommandInfo` and `getTopLevelCommandInfo`. Hope it's clear now.

File: source/argparse/internal/commandinfo.d [hunk 2]
```diff
@@ -50,18 +50,43 @@ unittest
...
+
+unittest
+{
+    @CommandInfo()
+    struct T {}
 
-    assert(name == "" || info.names.length > 0 && info.names[0].length > 0, "Command "~COMMAND.stringof~" must have name");
-    return info;
+    auto info = getSubCommandInfo!T(Config.init);
+    import std.stdio;stderr.writeln(info);
```
Comment (SirNickolas) [inline source/argparse/internal/commandinfo.d:89]: Accidentally left after a debugging session?

File: source/argparse/internal/commandinfo.d (hunk 1 above)
Comment (SirNickolas) [reply source/argparse/internal/commandinfo.d]: Oh, right. Yes, it became clear now, when `name` parameter is gone.

**Review by SirNickolas — APPROVED** [review]

File: source/argparse/internal/commandinfo.d (hunk 2 above)
Comment (andrey-zherikov) [reply source/argparse/internal/commandinfo.d]: yay! :-D
Removed

**Pull request merged**